certora/specs/Liveness.spec

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+import "Reverts.spec";
+
+// Check that having the allocator role allows to pause supply on the vault.
+rule canPauseSupply() {
+    env e1; MetaMorphoHarness.Id[] newSupplyQueue;
+    require newSupplyQueue.length == 0;
+    require e1.msg.value == 0;
+    require hasAllocatorRole(e1.msg.sender);
+
+    setSupplyQueue@withrevert(e1, newSupplyQueue);
+    assert !lastReverted;
+
+    storage pausedSupply = lastStorage;
+
+    env e2; uint256 assets2; address receiver2;
+    require assets2 != 0;
+    deposit@withrevert(e2, assets2, receiver2) at pausedSupply;
+    assert lastReverted;
+
+    env e3; uint256 shares3; address receiver3;
+    uint256 assets3 = mint@withrevert(e3, shares3, receiver3) at pausedSupply;
+    require assets3 != 0;
```

Comment:
If you mint 0 shares it will not revert, see [this run](https://prover.certora.com/output/66603/35dd11480343491c99c7de18051f5b7c?anonymousKey=2298c7865ca09dfae1f2ba4648f904ccfc0eae48). I assume this is because the conversion will give 0 assets (without rounding error), so the supplyMorpho is essentially skipped

Reply:
ok, didn't know that our 4626 doesn't revert on mint(0) (and probably deposit(0) neither then?)

Reply:
Right, I don't think `deposit(0)` reverts either